dmff/mbar.py

```diff
@@ -0,0 +1,194 @@
+import numpy as np
+import mdtraj as md
+from pymbar import MBAR
+import jax
+import jax.numpy as jnp
+from jax import grad
+from tqdm import tqdm, trange
+
+
+class BaseState:
+    def __init__(self, temperature, name):
+        self._temperature = temperature
+        self.name = name
+
+    def calc_energy_frame(self, frame):
```

Comment:
Following the convention of JAX, this function is better to be passed as an argument in the constructor.

Reply:
For Jax potential, we can use this way. For external MD engines, we'd better create new class for customize initialization of MD engine.
The design is updated in the latest version.